detection_rules/eswrap.py

```diff
@@ -736,3 +741,103 @@ def delete_ml_job(ctx: click.Context, job_name, job_type, verbose=True):
+                    "dns-a": {"type": "keyword"},
+                    "dns-aaaa": {"type": "keyword"},
+                    "dns-mx": {"type": "keyword"},
+                    "dns-ns": {"type": "keyword"},
+                    "banner-http": {"type": "keyword"},
+                    "fuzzer": {"type": "keyword"},
+                    "original-domain": {"type": "keyword"},
+                    "@timestamp": {"type": "date"},
+                    "dns.question.registered_domain": {"type": "keyword"}
+                }
+            }
+        }
+        return mappings
+
+    es_client.indices.create(index=f'dnstwist-{domain}', body=create_mappings())
+
+    es_updates = []
+    count = 0
+    for record in dnstwist_data:
+        temp = {}
+        temp['fuzzer'] = record.get('fuzzer', None)
+        if temp['fuzzer'] == 'original*':
+            continue
+        temp['dns-a'] = record.get('dns-a', None)
+        temp['dns-aaaa'] = record.get('dns-aaaa', None)
+        temp['dns-mx'] = record.get('dns-mx', None)
+        temp['dns-ns'] = record.get('dns-ns', None)
+        temp['banner-http'] = record.get('banner-http', None)
+        temp['original-domain'] = original_domain
+        temp['@timestamp'] = datetime.utcnow()
+        temp['dns'] = {'question': {}}
+        temp['dns']['question']['registered_domain'] = record.get('domain-name', None)
+        es_updates.append({'_index': domain_index, '_id': count, '_source': temp})
+        count += 1
+
+    click.echo(f'Indexing data for domain {original_domain}')
+
+    for success, info in helpers.streaming_bulk(es_client, es_updates, chunk_size=1000, request_timeout=150):
+        if not success:
+            client_error(info)
+
+    # Creating threat match rule
+    root_dir = Path(__file__).parent.parent
+    rule_template_file = root_dir / 'etc' / 'rule_template_typosquatting_domain.toml'
+    custom_rule_file = (
+        root_dir / "rules" / "cross-platform" / "initial_access_dns_request_for_typosquatting_domain.toml"
+    )
+
+    click.echo(f'{len(es_updates)} watchlist domains identified')
+
+    click.echo('Attempting to create threat match rule')
+    template_rule = load_dump(str(rule_template_file))
+
+    # Populate template rule with user's custom info
+    template_rule['metadata']['creation_date'] = datetime.today().strftime("%Y/%m/%d")
+    template_rule['metadata']['updated_date'] = datetime.today().strftime("%Y/%m/%d")
+    template_rule['rule']['author'].append(click.prompt('Enter rule author'))
+    template_rule['rule']['rule_id'] = str(uuid.uuid4())
+
+    # Create rule object and validate it against schema
+    rule_collection = RuleCollection()
+    rule_collection.load_dict(template_rule, path=custom_rule_file)
+
+    # Save rule in toml format
+    click.echo(f'Saving rule to {custom_rule_file}')
+    rule_collection.rules[0].save_toml()
+
+    if click.confirm("Upload rule to Kibana?", abort=True):
+        upload_rule(ctx, [custom_rule_file])
```

Comment:
@brokensound77 I can't seem to get this last step to execute successfully and upload the rule to Kibana. I just tried `ctx.invoke(upload_rule, [rule_collection.rules])`, but I see the error below. Any ideas?

```
Attempting to load dnstwist data from /Users/threatpunter/dnstwist_results.json
2006 records loaded
Original domain name identified: elastic.co
dnstwist index dnstwist-elastic already exists for elastic.co. Do you want to continue? [y/N]: >? y
Indexing data for domain elastic.co
2005 watchlist domains identified
Attempting to create threat match rule
Enter rule author: >? threatpunter
Saving rule to /Users/threatpunter/Documents/GitHub/detection-rules/rules/cross-platform/initial_access_dns_request_for_typosquatting_domain.toml
Upload rule to Kibana? [y/N]: >? y
CLI Error: Required: at least one of --rule-id, --rule-file, or --directory
```

Reply:
with `ctx.invoke` you need to specify keyword arguments. Also, `upload-rule` loads a rule from disk based on filename or ID, so you unfortunately can't pass a pre-existing collection in its current state. Instead, just pass the existing path.

try this:
```python
ctx.invoke(upload_rule, rule_file=custom_rule_file)
```